### PR TITLE
Discovery monitor table improvements

### PR DESCRIPTION
**Pull request by jpople**
Closes PROD-2197, PROD-2146

### Description Of Changes

Various additions/tweaks to discovery monitor table:

Screenshot 2024-07-09 at 00 53 31

* Add new columns:
    * Monitor frequency
    * Last monitored
    * Status
* Add new actions to "Actions" column:
    * Delete monitor
    * Execute ("scan") monitor
* Changes default monitor frequency from "not scheduled" to monthly 

### Code Changes

* [ ] Regenerate OpenAPI types
* [ ] Add new queries for deleting and executing monitors and new buttons to ActionCell
* [ ] Add new MonitorConfigEnableCell for Status column

### Steps to Confirm

* [ ] Visit /integrations
* [ ] Create or use an existing integration with valid credentials
* [ ] Create a discovery monitor
* [ ] Toggle "enabled" in the monitor's status cell on and off
* [ ] Click "scan" button; monitor should scan (note "last scanned" column doesn't update immediately as it's the timestamp that the scan _completes_)
* [ ] Delete the monitor

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* [ ] Issue Requirements are Met
* [ ] Update `CHANGELOG.md`

### CODE REVIEW

File: clients/admin-ui/src/features/integrations/configure-monitor/MonitorConfigEnableCell.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+export const MonitorConfigEnableCell = ({
+  row,
+  getValue,
+}: CellContext<MonitorConfig, boolean | undefined>) => {
+  const [putMonitor] = usePutDiscoveryMonitorMutation();
+
+  const onToggle = async (toggleValue: boolean) =>
+    putMonitor({
+      ...row.original,
+      last_monitored: undefined,
```
Comment (jpople) [inline clients/admin-ui/src/features/integrations/configure-monitor/MonitorConfigEnableCell.tsx:20]: `last_monitored` is read-only and (doesn't currently but) will throw an error if it's sent to the backend.  See [here](https://ethyca.atlassian.net/browse/PROD-2145?focusedCommentId=41803) for context.

File: clients/admin-ui/src/features/integrations/configure-monitor/ConfigureMonitorForm.tsx [hunk 2]
```diff
@@ -70,6 +70,7 @@ const ConfigureMonitorForm = ({
       ? {
           ...monitor,
           ...executionInfo,
+          last_monitored: undefined,
```
Comment (jpople) [inline clients/admin-ui/src/features/integrations/configure-monitor/ConfigureMonitorForm.tsx:73]: `last_monitored` is read-only and (doesn't currently but) will throw an error if it's sent to the backend.  See [here](https://ethyca.atlassian.net/browse/PROD-2145?focusedCommentId=41803) for context.

File: clients/admin-ui/src/features/data-discovery-and-detection/discovery-detection.slice.ts [hunk 3]
```diff
@@ -69,7 +73,10 @@ const discoveryDetectionApi = baseApi.injectEndpoints({
       query: (body) => ({
         method: "PUT",
         url: `/plus/discovery-monitor`,
-        body,
+        body: {
+          ...body,
+          last_monitored: undefined,
```
Comment (jpople) [inline clients/admin-ui/src/features/data-discovery-and-detection/discovery-detection.slice.ts:78]: `last_monitored` is read-only and shouldn't be sent in a request, see [https://ethyca.atlassian.net/browse/PROD-2145?focusedCommentId=41803](here) for context.

File: CHANGELOG.md [hunk 4]
```diff
@@ -114,16 +132,19 @@ The types of changes are:
 ## [2.38.1](https://github.com/ethyca/fides/compare/2.38.0...2.38.1)
 
 ### Changed
+
```
Comment (eastandwestwind) [inline CHANGELOG.md:135]: are these all intentional?

File: clients/admin-ui/src/features/data-discovery-and-detection/discovery-detection.slice.ts (hunk 3 above)
Comment (eastandwestwind) [reply clients/admin-ui/src/features/data-discovery-and-detection/discovery-detection.slice.ts:78]: could we actually make this a code comment?

File: clients/admin-ui/src/features/integrations/configure-monitor/ConfigureMonitorForm.tsx [hunk 5]
```diff
@@ -102,7 +102,7 @@ const ConfigureMonitorForm = ({
     name: monitor?.name ?? "",
     execution_start_date: format(initialDate, "yyyy-MM-dd'T'HH:mm"),
     execution_frequency:
-      monitor?.execution_frequency ?? (NOT_SCHEDULED as FrequencyOption),
+      monitor?.execution_frequency ?? MonitorFrequency.MONTHLY,
```
Comment (eastandwestwind) [inline clients/admin-ui/src/features/integrations/configure-monitor/ConfigureMonitorForm.tsx:105]: 👍

**Review by eastandwestwind — CHANGES_REQUESTED** [review]
Overall this is looking good @jpople . 

I ran the manual testing steps and things are functioning as expected. I want to point out one UI / usability issue, which is the "Actions" column is small for me, and cuts off some actions. I can horizontally scroll within that column but it seems we could instead make this column wider by default. 

Screenshot 2024-07-16 at 10 46 50 AM

Let me know your thoughts on this, and if you're able to replicate.

**Comment by jpople** [comment]
@eastandwestwind Thanks Catherine.  Removing the size constraint on the action column prevents the cell from becoming too small for the contents and now just cuts off the table (user can scroll the whole page horizontally) if it gets too small to hold everything, which seems better to me.

Screenshot 2024-07-16 at 11 54 13

**Comment by eastandwestwind** [comment]
Starting CR again...

**Review by eastandwestwind — APPROVED** [review]
@jpople , the `Actions` column looks a lot better, thanks!

Screenshot 2024-07-17 at 4 16 47 PM

**Pull request merged**